### ISSUE TITLE
[suggest] Change the IMPORTED target name to `libssh2::libssh2` or `Libssh2::Libssh2`.

### ISSUE THREAD

**Issue by hwhsu1231** [libssh2/libssh2#731]
If the **IMPORTED** target name is provided in form of `<pkg>::<pkg>`, I think it is better to keep the case of letters `<pkg>` the **SAME**, which means `libssh2::libssh2` or `Libssh2::Libssh2`.

For example:

- [`fmt::fmt`](https://github.com/fmtlib/fmt/blob/9.0.0/CMakeLists.txt#L232)
- [`spdlog::spdlog`](https://github.com/gabime/spdlog/blob/v1.10.0/CMakeLists.txt#L164)
- [`CLI11::CLI11`](https://github.com/CLIUtils/CLI11/blob/v2.2.0/CMakeLists.txt#L156)
- [`Catch2::Catch2`](https://github.com/catchorg/Catch2/blob/v3.1.0/src/CMakeLists.txt#L296)

Therefore, 

-   **Scheme 1:** `libssh2::libssh2`, modify:

    [L400 of src/CMakeLists.txt](https://github.com/libssh2/libssh2/blob/libssh2-1.10.0/src/CMakeLists.txt#L400)

    ```cmake
    install(EXPORT Libssh2Config
      NAMESPACE libssh2::
      DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libssh2)
    ```

    [L405 of src/CMakeLists.txt](https://github.com/libssh2/libssh2/blob/libssh2-1.10.0/src/CMakeLists.txt#L405)

    ```cmake
    export(TARGETS libssh2 NAMESPACE libssh2:: FILE Libssh2Config.cmake)
    ```

-   **Scheme 2:** `Libssh2::Libssh2`, modify:

    [L218 of src/CMakeLists.txt](https://github.com/libssh2/libssh2/blob/libssh2-1.10.0/src/CMakeLists.txt#L218)

    ```cmake
    add_library(Libssh2 ${SOURCES})
    ```

How about that?

**Comment by willco007** [comment]
I'd probably vote for lower case; how does this impact folks using a pre-build library?

**Comment by jzakrzewski** [comment]
@hwhsu1231 shouldn't the line 406 also be changed to lower case? Also, if upper casing, the output file would probably have to be renamed, otherwise you'll get "Libssh2.so", which would wreak havoc.

@willco007 the changes will only impact CMake users that previously used "Libssh2::libssh2" imported target.

**Comment by willco007** [comment]
@bagder What do you think about this change?

**Comment by hwhsu1231** [comment]
@jzakrzewski 

> shouldn't the line 406 also be changed to lower case? Also, if upper casing, the output file would probably have to be renamed, otherwise you'll get "Libssh2.so", which would wreak havoc.

Yes, I think so if the schema 1 is adopted.

> the changes will only impact CMake users that previously used "Libssh2::libssh2" imported target.

How about prepare the _old_ target and the _new_ one simultaneously for a while? And then, add some comments in the source code to tell users that the old one will be deprecated in the future?

**Comment by hwhsu1231** [comment]
However, considering the backward compatibility, maybe it's better to keep the current imported target name.

**Comment by vszakats** [comment]
Here's a patch proposal for the current master, in case we want to pursue this:
```diff
diff --git a/docs/INSTALL_CMAKE.md b/docs/INSTALL_CMAKE.md
index fe203068..47ca7318 100644
--- a/docs/INSTALL_CMAKE.md
+++ b/docs/INSTALL_CMAKE.md
@@ -168,8 +168,8 @@ setting it up with libssh2's location.  Just add just the following
 lines and CMake will find libssh2 on your system, set up the necessary
 paths and link the library with your binary.
 
-    find_package(Libssh2 REQUIRED CONFIG)
-    target_link_libraries(my_project_target Libssh2::libssh2)
+    find_package(libssh2 REQUIRED CONFIG)
+    target_link_libraries(my_project_target libssh2::libssh2)
 
 Of course, you still have to make libssh2 available on your system
 first.  You can install it in the traditional way shown above, but you
@@ -184,13 +184,13 @@ builds your project:
     include(ExternalProject)
 
     ExternalProject_Add(
-        Libssh2
+        libssh2
         URL <libssh2 download location>
         URL_HASH SHA1=<libssh2 archive SHA1>
         INSTALL_COMMAND "")
 
     ExternalProject_Add(
-        MyProject DEPENDS Libssh2
+        MyProject DEPENDS libssh2
         SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src
         INSTALL_COMMAND "")
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c35237c8..8ebdad94 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -457,13 +457,13 @@ set(RUNTIME_DEPENDENCIES ${_RUNTIME_DEPENDENCIES} CACHE INTERNAL
 
 ## During package installation, install Libssh2Config.cmake
 install(EXPORT Libssh2Config
-  NAMESPACE Libssh2::
+  NAMESPACE libssh2::
   DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libssh2)
 
 ## During build, register directly from build tree
 # create Libssh2Config.cmake
-export(TARGETS ${libssh2_export} NAMESPACE Libssh2:: FILE Libssh2Config.cmake)
-export(PACKAGE Libssh2) # register it
+export(TARGETS ${libssh2_export} NAMESPACE libssh2:: FILE Libssh2Config.cmake)
+export(PACKAGE libssh2)  # register it
 
 ## Export a .pc file for client projects not using CMaek
 if(PC_REQUIRES_PRIVATE)
```

**Comment by vszakats** [comment]
We won't do this now to retain compatibility, but would welcome a patch that would be exporting both names simultaneously, as suggested [here](https://github.com/libssh2/libssh2/issues/731#issuecomment-1377573211).

**Comment by vszakats** [comment]
This was implemented in 2da13c13e14d990df23686490a1e98d963ea7f71 while also keeping compatibility. ('Restoring' would be more precise as compatibility broke in v1.11.0 due to other changes).
